### ISSUE TITLE
New Blog Design

### ISSUE THREAD

**Issue by steven1046** [OpenLiberty/openliberty.io#458]
Blog page redesign where clicking on a blog will open the blog above the all blogs page so the user can close the blogs easily (using the 'x' or clicking behind the blog in the blue sidebar behind the blog) and continue reading. Each blog will have tags and you will be able to sort the blogs page by tags. 

Design as of Jan 27 2019
https://ibm.box.com/s/90f47zu2iub7v1nlsu8bpz7328wrfn7m

All Blogs page:
![image](https://user-images.githubusercontent.com/6392944/42907719-4d2ec1b8-8aa4-11e8-8884-52e4c6feb1c6.png)

Pop-up blog with the all blogs page behind it:
![image](https://user-images.githubusercontent.com/6392944/42907837-ab6f17f0-8aa4-11e8-87c0-1fc9e64033de.png)

End of the blog:
![image](https://user-images.githubusercontent.com/6392944/42908045-655d7be8-8aa5-11e8-9893-3232359273e4.png)

**Comment by kinueng** [comment]
Confirmed that we need to discuss the latest design as a group before we start implementation.

**Comment by steven1046** [comment]
I don't think tags should both at the beginning of a blog and at the end. It makes the most sense to put them at the beginning.

**Comment by cajaygle** [comment]
For blog tags we'll use 'event', 'release', and the same content tags we're using for guides.  Tags to be listed here by @yeekangc  https://github.com/OpenLiberty/blogs

**Comment by cajaygle** [comment]
We'll tag blog posts with 'JakartaEE' but in the Featured Tags list in the blog page sidebar we'll list it as 'JakartaEE/JavaEE'.  Site search should find either.

For the Featured Tags list in the sidebar, @lauracowen can you revisit this after we have all the current blog posts tagged.  We can decide whether we want to list all tags here or only a subset as Featured Tags.

**Comment by yeekangc** [comment]
Current tags for guides: microprofile, maven, docker, kubernetes, java ee, security

**Comment by cajaygle** [comment]
<img width="786" alt="screen shot 2019-01-31 at 1 36 25 pm" src="https://user-images.githubusercontent.com/27516391/52077286-a5c7d080-255e-11e9-9bcd-ecae21e0b0e7.png">

Mockup showing how we'll link to third-party blog posts.  Format may be "Continue reading on ibm.com", or "Continue reading on John Smith's blog" if more appropriate.

**Comment by cajaygle** [comment]
For older blog posts we'll use a more compact view to display them on the blog page.   Design forthcoming.

**Comment by cajaygle** [comment]
Opened issue #973 for some tweaks to the blog page design.

**Comment by steven1046** [comment]
Jay said in https://github.com/OpenLiberty/openliberty.io/issues/973#issue-407882162: In the new blog design, if a user comes to a blog post from an external link it's not clear how they navigate to other blog posts or the rest of the site.

We should keep the OL site navigation at the top of the page, as well as a link back to the main blog page. Currently navigation links are presented at the bottom of the blog page in the new design.

The Share action could be moved to the top as well (or put at both top and bottom).

**Comment by mdenby** [comment]
Here are the designs that Jay referenced in #973 with the navigation updated at the top, removed the 'X', added a 'Back to blog posts' at the top and bottom, and the share buttons at the top and bottom too. This shows the desktop, tablet, and mobile views.

![7_blog_post_desktop_top](https://user-images.githubusercontent.com/26823018/52733902-ea596000-2f91-11e9-8134-effb41526349.png)
![8_blog_post_desktop_bottom](https://user-images.githubusercontent.com/26823018/52733903-ea596000-2f91-11e9-965c-1369d7432c1a.png)
![10_blog_post_mobile](https://user-images.githubusercontent.com/26823018/52733904-ea596000-2f91-11e9-9524-1fc5441a0dd5.png)
![9_blog_post_tablet](https://user-images.githubusercontent.com/26823018/52733905-ea596000-2f91-11e9-9126-a3016e0c713e.png)

**Comment by mbroz2** [comment]
No "Recommended Posts" at the end of the blog.  No "Featured Blogs" section (no pinning of blogs)

**Comment by TreeHanh** [comment]
Blogs page design with compact view of older posts. 

![compact blogs](https://user-images.githubusercontent.com/18586040/52793273-5b088700-303b-11e9-9ee0-8c2e5cf574a6.jpg)

**Comment by cajaygle** [comment]
Notice as part of the compact view of older blog posts that we updated how third-party blog posts are displayed.  The third-party designation (e.g. "[on CNET.com]") is now displayed after the title.  This allows a consistent presentation for full cards and compact cards.  Also, the **Read more** button color is orange to signify an external link (opened in a new window), which is our style throughout the site.

**Comment by mdenby** [comment]
@cajaygle Here is an updated version of the Blog page with 'Recommended Posts' removed. Does this look okay? 

![8_blog_post_desktop_bottom](https://user-images.githubusercontent.com/26823018/52809639-be57e080-305e-11e9-936f-24cfaf736abf.png)
![10_blog_post_mobile](https://user-images.githubusercontent.com/26823018/52809640-be57e080-305e-11e9-8d63-adb78107b823.png)
![9_blog_post_tablet](https://user-images.githubusercontent.com/26823018/52809642-be57e080-305e-11e9-999a-9b2fbc37ead3.png)

**Comment by cajaygle** [comment]
👍 Back link at the top of page and back button at bottom of page should both say "See all blog posts".

**Comment by TreeHanh** [comment]
This is the design for the new blog page. It now shows the "latest posts" which is followed by a compacted view of "older posts". There are some changes in language, and we've also added an RSS feed on the left and also at the end of the blog posts (see second picture). The footer is sized down under the blog post section and will expand up when hovered on (also see second picture). 
![blog_page](https://user-images.githubusercontent.com/18586040/53249060-23c25780-3685-11e9-99e9-c5e451bc98f5.jpg)

![blog_page_footer_up](https://user-images.githubusercontent.com/18586040/53249068-2d4bbf80-3685-11e9-89ec-9ac58583d5e2.jpg)

**Comment by cajaygle** [comment]
I know we've been dribbling out blog design changes.  Thuy-Hanh's changes should be the last.  I thought it was worth adding the RSS feed link after Mike Thompson talked about it.  The wording on the left changed to include Twitter and RSS, and likewise the wording at the end of the righthand column.

Because the list of blog posts may be long, we present a compact footer on the right side at all times.  Then as Thuy-Hanh noted, on hover it expands to the full footer (but in the righthand column only).  If we end up with other pages on the site that have long scrolling content we can employ the same compact footer.

**Comment by kinueng** [comment]
Tagging @lauracowen so she can see the latest blog designs.

**Comment by lauracowen** [comment]
Looks good. I don't see any obvious problems or omissions.

**Comment by lauracowen** [comment]
Stylesheet comment:

In the current live blog stylesheet, the 'definition list' is more distinct than any of our sub-headings. Which seems wrong. I know there are plans in the design above to fix the sub-headings (H2 and H3 in particular need to be more much prominent than in the live site currently) but maybe the definition list entries should be lighter weight (maybe indented instead of being so bold?):

![image](https://user-images.githubusercontent.com/3322532/55616966-f3dd9800-578a-11e9-8d3a-812feb4030d4.png)

Also, if possible, if the implementation work is broken down into chunks, could fixing the headings be one of the early things to do and applied to the existing live design?

**Comment by ellenwyllie** [comment]
I've been working on implementing these redesigns and I am 99% done. So as soon as I hear back from design I can fix the heading issues. The redlines don't have specifications for the heading sizes so I'm not entirely sure how we should fix these problems. @cajaygle @TreeHanh Could we get details about the new heading styling and definition list styling?

**Comment by ellenwyllie** [comment]
We've been discussing some changes to the "Latest Posts" section. There is concern that showing the 3 latest posts will make posts look old even if they are not. We discussed displaying latest posts based on the date rather than always showing the 3 latest. But we will show a minimum of 3 just in case there is a period without any recent posts.

**Comment by lauracowen** [comment]
I agree with @NottyCode 's point on Slack that the yellow should be green like elsewhere.
I'm happy with the design in general but the headings and definition lists styling need to be sorted out. I had a discussion on Slack with @TreeHanh which I've just realised I didn't finish replying to the last suggestion (sorry) so I've just commented on there. The lack of prominence in the headings relative to the rest of the post body and styling is a big problem for readability at the moment (in the live site design) so I want to get this fixed before we release this new blog design.

**Comment by ellenwyllie** [comment]
@lauracowen Have you looked at the demo site recently? I changed the beam to green like Alasdair requested (You might need to clear your cache to see the updated image). The heading and definition list styling has also been updated and can be seen on the demo site: https://demo1-openlibertyio.mybluemix.net/blog/

**Comment by lauracowen** [comment]
Another comment (maybe for Design - sorry) is about the bullet points vertical spacing. See https://openliberty.io/blog/2019/03/28/microprofile22-liberty-19003.html

They're quite widely spread and I think could be a little more compact but I'd appreciate Design input on it. I'm happy to raise this as a separate issue if you'd rather not hold up the new blog design (I don't think it's critical).

**Comment by mbroz2** [comment]
@lauracowen we currently do not localize (nor translate) any content on the site.  If that is a request for the site, we'll want to address it in a separate issue.  In the meantime, I suggest blog authors use the written month so that it's clear what the date is, regardless of the format (aka, what that blog is already doing)

**Comment by TreeHanh** [comment]
@lauracowen That is easy fix. I've talked with Ellen on making the spacing between the bullet points smaller.

**Comment by ellenwyllie** [comment]
So I indented the definition lists, made the h2 bold so that it stands out more, changed the orange links to a blue/purple color on blog posts and decreased the spacing between bullet points.

The footer is no longer fixed to the bottom of the screen. After discussions on the site call this morning and more talks with design afterwards, we decided not to implement the hovering behavior for the footer. So the footer is back to the way it used to be. If it becomes an issue in the future, we will revisit this.
